### PR TITLE
build CKEditor5

### PR DESCRIPTION
**Pull request by WebberMao**
Hello, I build CKEditor5 in this RP.

The user can use ckeditor4 or ckeditor5 by choosing different CKEditor class
`app = CKEditor4(app)  #for ckeditor4`
`app = CKEditor5(app)  #for ckeditor5`

and change the editor_type (support CKEditor5 only) by setting config
`CKEDITOR_EDITOR_TYPE = 'classic'  #and all of the following

- classic for Classic Editor
- inline for Inline Editor
- balloon for Balloon Editor
- balloon-block for Balloon Block Editor
- decoupled-document for Decoupled Editor

hope you will like it!

### CODE REVIEW

File: flask_ckeditor/__init__.py [hunk 1]
```diff
@@ -163,7 +163,82 @@ def load_code_theme():
...
+
+        return Markup('<script src="%s"></script>' % url)
+
+    def config(self, name='ckeditor', custom_config='', **kwargs):
+        """Config CKEditor."""
+
+
+        file_uploader = kwargs.get('file_uploader', current_app.config['CKEDITOR_FILE_UPLOADER'])
+
+        if file_uploader != '':
```
Comment (abdounasser202) [inline flask_ckeditor/__init__.py:210]: Why not something like `if not file_uploader` instead of `if file_uploader != ''` ?

Comment (abdounasser202) [reply flask_ckeditor/__init__.py:210]: By the way, nice work, @WebberMao

**Comment by malikpiara** [comment]
Hi @WebberMao! Super excited to learn CKEditor5 might be available soon! Any news on the merge? Also: Does it make sense to add GitHub sponsors to this project?

**Comment by WebberMao** [comment]
> Hi @WebberMao! Super excited to learn CKEditor5 might be available soon! Any news on the merge? Also: Does it make sense to add GitHub sponsors to this project?

**Comment by WebberMao** [comment]
@malikpiara I'm not sure. Maybe there are some test before merging.  I am sorry that I can't answer these questions properly.  You should ask the maintainer @greyli  if he is free.

File: flask_ckeditor/__init__.py [hunk 2]
```diff
@@ -163,7 +163,82 @@ def load_code_theme():
...
+        })                                 
+        .then( ckeditor => {                                                                
+                window.ckeditor = ckeditor;                                                 
+                const toolbarContainer = document.querySelector( '#ckeditor-toolbox' );
+                toolbarContainer.appendChild( ckeditor.ui.view.toolbar.element );
+        })
+        .catch( error => {
+                console.error( error );
+        });
+        document.querySelector( '#submit' ).addEventListener( 'click', () => {
```
Comment (greyli) [inline flask_ckeditor/__init__.py:229]: What if the user uses a different id for the submit button?

**Comment by greyli** [comment]
Hi, thanks for working on this. I think this PR is still incomplete:

- We shouldn't introduce a breaking change directly. The `CKEditor` class should be kept with a deprecated warning.
- The CKEditor5 support for some configurations is not implemented (e.g. `CKEDITOR_SERVE_LOCAL`, `CKEDITOR_LANGUAGE`, etc.)
- The example application, tests, docs, and changelog should be updated.

**Comment by greyli** [comment]
Since CKEditor has changed the license to GPL, we decided not to add support to CKEditor 5. Thank you for working on this feature!

**Pull request closed without merging**